File: app/src/main/java/wangdaye/com/geometricweather/db/entities/MinutelyEntity.java
package wangdaye.com.geometricweather.db.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.Index;
import java.util.Date;
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode;

/**
 * Minutely entity matching GreenDAO schema 62.
 */
@Entity(tableName = "MINUTELY_ENTITY",
        indices = {@Index(value = {"CITY_ID", "WEATHER_SOURCE"})})
public class MinutelyEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ID")
    public Long id;

    @ColumnInfo(name = "CITY_ID")
    public String cityId;

    @ColumnInfo(name = "WEATHER_SOURCE")
    public String weatherSource;

    @ColumnInfo(name = "DATE")
    public Date date;

    @ColumnInfo(name = "TIME")
    public long time;

    @ColumnInfo(name = "DAYLIGHT")
    public boolean daylight;

    @ColumnInfo(name = "WEATHER_TEXT")
    public String weatherText;

    @ColumnInfo(name = "WEATHER_CODE")
    public WeatherCode weatherCode;

    @ColumnInfo(name = "MINUTE_INTERVAL")
    public int minuteInterval;

    @ColumnInfo(name = "DBZ")
    public Integer dbz;

    @ColumnInfo(name = "CLOUD_COVER")
    public Integer cloudCover;

    public MinutelyEntity() {
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/entities/MinutelyEntity.kt
package wangdaye.com.geometricweather.db.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.Date
import kotlin.jvm.JvmField
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode

/**
 * Minutely entity matching GreenDAO schema 62.
 */
@Entity(
    tableName = "MINUTELY_ENTITY",
    indices = [Index(value = ["CITY_ID", "WEATHER_SOURCE"])]
)
class MinutelyEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ID")
    @JvmField
    var id: Long? = null

    @ColumnInfo(name = "CITY_ID")
    @JvmField
    var cityId: String? = null

    @ColumnInfo(name = "WEATHER_SOURCE")
    @JvmField
    var weatherSource: String? = null

    @ColumnInfo(name = "DATE")
    @JvmField
    var date: Date? = null

    @ColumnInfo(name = "TIME")
    @JvmField
    var time: Long = 0L

    @ColumnInfo(name = "DAYLIGHT")
    @JvmField
    var daylight: Boolean = false

    @ColumnInfo(name = "WEATHER_TEXT")
    @JvmField
    var weatherText: String? = null

    @ColumnInfo(name = "WEATHER_CODE")
    @JvmField
    var weatherCode: WeatherCode? = null

    @ColumnInfo(name = "MINUTE_INTERVAL")
    @JvmField
    var minuteInterval: Int = 0

    @ColumnInfo(name = "DBZ")
    @JvmField
    var dbz: Int? = null

    @ColumnInfo(name = "CLOUD_COVER")
    @JvmField
    var cloudCover: Int? = null
}
